tests/tests/src/json_decoders.mjs

```diff
@@ -0,0 +1,95 @@
+// Generated by ReScript, PLEASE EDIT WITH CARE
+
+import * as $$Array from "rescript/lib/es6/Array.js";
+
+function decodeUser(json) {
+  if (json === null) {
+    return;
+  }
+  if (!(typeof json === "object" && !Array.isArray(json))) {
+    return;
+  }
+  let match = json.id;
+  if (match === undefined) {
+    return;
+  }
+  if (match === null) {
+    return;
+  }
+  if (typeof match !== "string") {
+    return;
+  }
+  let match$1 = json.name;
+  if (match$1 === undefined) {
+    return;
+  }
+  if (match$1 === null) {
+    return;
+  }
+  if (typeof match$1 !== "string") {
+    return;
+  }
+  let match$2 = json.age;
+  if (match$2 === undefined) {
+    return;
+  }
+  if (match$2 === null) {
+    return;
+  }
+  if (typeof match$2 !== "number") {
+    return;
+  }
+  let email = json.email;
+  let tmp;
+  tmp = email !== undefined && !(email === null || typeof email !== "string") ? email : undefined;
```

Comment:
This one is similar to the above - `typeof email !== "string" ? undefined : email` or even `typeof email === "string" ? email : undefined`.